src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -81,70 +67,27 @@ func (r *ClusterReconciler) Reconcile(
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var svc corev1.Service
-
-	err := r.Get(ctx, types.NamespacedName{Name: redpandaCluster.Name, Namespace: redpandaCluster.Namespace}, &svc)
-	if err != nil && !errors.IsNotFound(err) {
-		log.V(debugLevel).Info("Unable to fetch Service resource")
-		return ctrl.Result{}, err
-	}
-
-	if errors.IsNotFound(err) {
-		log.V(debugLevel).Info("Creating headless service")
-
-		if err = r.createHeadlessService(ctx, &redpandaCluster, r.Scheme); err != nil {
-			log.Error(err, "Failed to create new service",
-				"Service.Namespace", redpandaCluster.Namespace,
-				"Service.Name", redpandaCluster.Name)
-
-			return ctrl.Result{}, err
-		}
-	}
-
-	var seedConfigMap corev1.ConfigMap
-
-	err = r.Get(ctx, types.NamespacedName{Name: redpandaCluster.Name + seedSuffix, Namespace: redpandaCluster.Namespace}, &seedConfigMap)
-	if err != nil && !errors.IsNotFound(err) {
-		log.V(debugLevel).Info("Unable to fetch seed redpanda ConfigMap resource")
-
-		return ctrl.Result{}, err
-	}
-
-	if errors.IsNotFound(err) {
-		log.V(debugLevel).Info("Creating seed redpanda ConfigMap")
-
-		if err = r.createBootstrapConfigMap(ctx, &redpandaCluster, r.Scheme); err != nil {
-			log.Error(err, "Failed to create new seed redpanda ConfigMap",
-				"Configmap.Namespace", redpandaCluster.Namespace,
-				"Configmap.Name", redpandaCluster.Name+seedSuffix)
-
-			return ctrl.Result{}, err
-		}
+	sts := resources.NewStatefulSet(r.Client, &redpandaCluster, r.Scheme)
+	resources := []resources.OwnedResource{
```

Comment:
The order here dictates execution order, does the order matter?

Reply:
I preserved the same order. Yeah I think at least configmap should go before sts always...